File: 2020/01.py
import numpy as np

instructions = np.loadtxt("2020/input_01.txt")

smaller = []
larger = []
for i in instructions:
    if i < 1010:
        smaller.append(i)
    else:
        larger.append(i)

for i in smaller:
    for j in larger:
        if i + j == 2020:
            # ANSWER 1
            print(i * j)
            break
    else:
        continue
    break

for i in range(len(instructions) - 2):
    for j in range(i + 1, len(instructions) - 1):
        for k in range(j + 1, len(instructions)):
            # can use np.sum(instructions[[i, j, k]]) (also np.prod(instructions[[i, j, k]])) but it's a lot slower lol
            if instructions[i] + instructions[j] + instructions[k] == 2020:
                # ANSWER 2
                print(instructions[i] * instructions[j] * instructions[k])
                break
        else:
            continue
        break
    else:
        continue
    break

File: 2020/01.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"strconv"
)

func main() {
    // open file
    f, err := os.Open("2020/input_01.txt")
    if err != nil {
        log.Fatal(err)
    }
    // remember to close the file at the end of the program
    defer f.Close()

	var input []int 

    // read the file line by line using scanner
    scanner := bufio.NewScanner(f)

    for scanner.Scan() {
        // do something with a line
		i, _ := strconv.Atoi(scanner.Text())
        input = append(input, i)
	}

    if err := scanner.Err(); err != nil {
        log.Fatal(err)
    }

	var smaller []int
	var larger []int

	for _, i := range input {
		if i < 1010 {
			smaller = append(smaller, i)
		} else {
			larger = append(larger, i)
		}
	}

	var entries [2]int

	for _, s := range smaller {
		for _, l := range larger {
			if s + l == 2020 {
				entries[0] = s
				entries[1] = l
				break
			}
		}
		if entries[0] != 0 {
			break
		}
	}

	// ANSWER 1
	fmt.Println(entries[0] * entries[1])

	var entries_trio [3]int

	for i := 0; i < len(smaller); i++ {
		first := smaller[i]
		for j := i; j < len(smaller); j++ {
			second := smaller[j]
			for k := j; k < len(smaller); k++ {
				third := smaller[k]

				if first + second + third == 2020 {
					entries_trio[0] = first
					entries_trio[1] = second
					entries_trio[2] = third
					break
				}
			}
			if entries_trio[0] == 0 {
				for _, l := range larger {
					if first + second + l == 2020 {
						entries_trio[0] = first
						entries_trio[1] = second
						entries_trio[2] = l
						break
					}
				}
			} 
			if entries_trio[0] != 0 {
				break
			}
		}
		if entries_trio[0] != 0 {
				break
		}
	}
	
	// ANSWER 2
	fmt.Println(entries_trio[0] * entries_trio[1] * entries_trio[2])
}